warp10/src/main/java/io/warp10/standalone/StandaloneDeleteHandler.java

```diff
@@ -380,7 +382,26 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
       drequest.setClassSelectors(clsSels);
       drequest.setLabelsSelectors(lblsSels);
 
-      metadatas = directoryClient.find(drequest);
+      metadatas = new ArrayList<Metadata>();
+
+      MetadataIterator iter = directoryClient.iterator(drequest);
+
+      //
+      // Apply token scope
+      //
+
+      if (writeToken.getAttributesSize() > 0) {
+        ReadToken rtoken = new ReadToken();
+        rtoken.setAttributes(writeToken.getAttributes());
+        rtoken.addToApps(writeToken.getAppName());
+        rtoken.addToOwners(writeToken.bufferForOwnerId());
+        rtoken.addToProducers(writeToken.bufferForProducerId());
+        iter = FIND.getScopedIterator(iter, rtoken, null);
+      }
```

Comment:
WriteToken should not have .nofind or .nofetch attributes, this would be an odd case which we are not taking into account here. Did you see any other potential issue besides those unlikely ones?

Reply:
I cannot understand why it works without copying the other token properties:
```
        rtoken.setTokenType(TokenType.READ);
        rtoken.setLabels(writeToken.getLabels());
        rtoken.setAppName(writeToken.getAppName());
        rtoken.setExpiryTimestamp(writeToken.getExpiryTimestamp());
```

Reply:
META is still not covered by tokenscope, by the way.